tests/unit/setup-application-context-test.js

```diff
@@ -29,8 +33,21 @@ module('setupApplicationContext', function(hooks) {
   if (!hasEmberVersion(2, 4)) {
     return;
   }
-
+  let isolatedApp;
   hooks.beforeEach(async function() {
+    const AppConfig = assign({ autoboot: false }, config.APP);
+    // .extend() because initializers are stored in the constructor, and we
```

Comment:
This seems somewhat unrelated to the rest of the changes, which failures caused you to need to change this?

Reply:
@AlexTraher and I found the `setup-application-context-test.js` tests to fail intermittently before introducing this.

After some digging we found it was order dependent, for instance when we ran these two tests:
>setupApplicationContext: can perform a basic template rendering
>DOM Helper: click > non-focusable element types: clicking a div via element with context set

If the "DOM Helper" test was run first then we found that the "setupApplicationContext" test failed, however if you reran the tests then the "setupApplicationContext" would go first and both tests would pass.

We found commenting out [this line](https://github.com/emberjs/ember-test-helpers/pull/482/files/a7db2583040cb566d0601bb95450a7d25f39ceac#diff-2556457e0cd69598faa308feee8b8ffaR51) in the `hasPendingTransitions` method. Stopped the error, but is needed for the logic to work.

We attempted to find the bug, thought we might have some state that was not being cleaned up correctly, but I am sad to say we were unsuccessful.

So the next best idea we had was to construct an isolated `Ember.App()` to run the failing tests against [as is done here](https://github.com/emberjs/ember-test-helpers/blob/50c24e181f03b511cb9cb94b357251160fc1496e/tests/unit/setup-context-test.js#L294) and thats how we came to this solution. Neither of us were particularly happy with this and we hope there is a better one if you have any ideas?